pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -184,9 +184,11 @@ func buildMixerOutboundTCPFilter(env *model.Environment, node *model.Proxy) list
 // buildHTTPMixerFilterConfig builds a mixer HTTP filter config. Mixer filter uses outbound configuration by default
 // (forward attributes, but not invoke check calls)  ServiceInstances belong to the Node.
 func buildHTTPMixerFilterConfig(mesh *meshconfig.MeshConfig, role model.Proxy, nodeInstances []*model.ServiceInstance, outboundRoute bool, config model.IstioConfigStore) *mccpb.HttpClientConfig { // nolint: lll
+	mcs, _, _ := net.SplitHostPort(mesh.MixerCheckServer)
+	mrs, _, _ := net.SplitHostPort(mesh.MixerReportServer)
 	transport := &mccpb.TransportConfig{
-		CheckCluster:  v1.MixerCheckClusterName,
-		ReportCluster: v1.MixerReportClusterName,
+		CheckCluster:  model.BuildSubsetKey(model.TrafficDirectionOutbound, "", mcs+".svc.cluster.local", &model.Port{Name: "tcp-plain"}),
+		ReportCluster: model.BuildSubsetKey(model.TrafficDirectionOutbound, "", mrs+".svc.cluster.local", &model.Port{Name: "tcp-plain"}),
```

Comment:
this is hacky.. We can tell people to use FQDN in the mesh config.. or you can use model.ResolveHostnametoFQDN to get the FQDN of mixer service (I prefer the former). And with regard to the port, i think you should be able to lookup the Mixer service from mesh.env.Services(), and from the mixer service, lookup the name for the mixer port that the user provided.